src/hasi/hasi/meshtomeshregistrar.py

```diff
@@ -39,39 +39,50 @@ def register(self, template_mesh:MeshType, target_mesh:MeshType,
         raise NotImplementedError('Base class does not implement register functionality')
 
     # Convert itk.Mesh object to 3-dimensional itk.Image object
-    # Defined for mesh of type itk.F
-    def mesh_to_image(self, mesh:MeshType, reference_image:ImageType=None) -> ImageType :
+    def mesh_to_image(self, meshes:list, reference_image:ImageType=None) -> ImageType :
+        # Allow single mesh as input
+        if type(meshes) == self.MeshType:
+            meshes = [meshes]
 
         MeshToImageFilterType = itk.TriangleMeshToBinaryImageFilter[self.MeshType,self.ImageType]
+        images = list()
 
         if not reference_image:
-            bounds = mesh.GetBoundingBox().GetBounds()
+            # Set bounds to largest region encompassing all meshes
+            # Bounds format: [x_min x_max y_min y_max z_min z_max]
+            bounds = meshes[0].GetBoundingBox().GetBounds()
+            for mesh in meshes[1:]:
+                mesh_bounds = mesh.GetBoundingBox().GetBounds()
+                for dim in range(0, self.Dimension):
+                    bounds[dim*2] = min(bounds[dim*2], mesh_bounds[dim*2])
+                    bounds[(dim*2)+1] = max(bounds[(dim*2)+1], mesh_bounds[(dim*2)+1])
+
+            # Calculate spacing, origin, and size with 5 pixel buffer around mesh
             spacing = ((bounds[1] - bounds[0]) / 90,
                         (bounds[3] - bounds[2]) / 90,
                         (bounds[5] - bounds[4]) / 90)
             origin = (bounds[0] - 5 * spacing[0],
                         bounds[2] - 5 * spacing[1],
                         bounds[4] - 5 * spacing[2])
             size = (100,100,100)
-        
+        else:
+            # Use given parameters
```

Comment:
Shouldn't we also use direction from reference image? If `TriangleMeshToBinaryImageFilter` supports direction, we should use the one from reference.

Reply:
Agreed. For our current use case where meshes are already aligned direction will always be the default identity matrix but this may not always be the case.